### ISSUE TITLE
Use the default GRADLE_USER_HOME if possible for Gradle wrappers and modules

### ISSUE THREAD

**Issue by JullsP** [redhat-developer/vscode-java#946]
Gradle wrappers and modules are always downloaded in `GRADLE_HOME` instead of the usual `GRADLE_USER_HOME` in Linux.

##### Environment
- Operating System: `Xubuntu 18.04`
- Gradle & JVM version: `5.4.1` & `1.8.0_212`
- Visual Studio Code version: `1.34.0`
- Java extension version: `0.46.0`

##### Steps To Reproduce
1. Using the locally installed Gradle distribution in `GRADLE_HOME`, create a simple project using `gradle init` (with, or without dependency modules in build script).
2. Open the project in Visual Studio Code.

##### Current Result
Gradle wrappers and dependency module caches are saved in `GRADLE_HOME` instead of `GRADLE_USER_HOME` directory.

##### Expected Result
Should use the default `GRADLE_USER_HOME` directory if possible, so modules and wrappers won't be downloaded twice in separate directories.

**Comment by fbricon** [comment]
@snjeza please investigate. Need to compare with default buildship behavior

**Comment by kaffeekonverter** [comment]
Might be useful when investigating: Unsetting the `GRADLE_HOME` environment variable seems to fix this issue (temporary). For recent gradle versions, the [installation instructions](https://gradle.org/install/#manually) doesn't mention the `GRADLE_HOME` variable at all.

**Comment by JullsP** [comment]
I just switched back to windows 10 and this issue still persist. At first, I was wondering why the language server is taking too long to initialize a basic gradle project, then I saw that it is downloading two gradle wrapper distribution, one in the user home directory (the `GRADLE_USER_HOME`), and the other one is in the local `GRADLE_HOME` directory.

I've wasted hours restarting vscode and searching for solution :expressionless:. I think a better fix for this is to rely on a locally installed gradle distribution first (i.e use the `GRADLE_HOME` distribution), if that fails, then download the gradle wrapper to the default user home directory (the `GRADLE_USER_HOME` where it should be) and never touch the directory of `GRADLE_HOME` environment variable coz it's just a waste of storage space.
